handlers/templates/pages/guest-link-create.html

```diff
@@ -102,6 +107,24 @@ <h1 class="title">Create Guest Link</h1>
       </div>
     </div>
 
+    <div class="field my-5">
+      <label class="label">Guest Files Expire</label>
+      <div class="control">
+        <div class="select">
+          <select id="expiration-select-file">
+            {{ range .ExpirationOptions }}
```

Comment:
I think this is tripping you up and making the logic confusing. When it's a list of expiration options, we have to awkwardly convert from `ExpirationTime` to `FileLifetime`. I think it's a lot cleaner to start with `FileLifetime` from the start.

I suggest updating `guestLinksNewGet` to something like this:

```go

func (s Server) guestLinksNewGet() http.HandlerFunc {
	fns := template.FuncMap{
		"formatExpiration": func(t time.Time) string {
			return t.Format(time.RFC3339)
		},
		"formatLifetime": func(l picoshare.FileLifetime) string {
			return l.Duration().String()
		},
	}

	t := parseTemplatesWithFuncs(fns, "templates/pages/guest-link-create.html")

	return func(w http.ResponseWriter, r *http.Request) {
		type expirationOption struct {
			FriendlyName string
			Expiration   time.Time
			IsDefault    bool
		}
		type fileLifetimeOption struct {
			FileLifetime picoshare.FileLifetime
			IsDefault    bool
		}
		if err := t.Execute(w, struct {
			commonProps
			ExpirationOptions   []expirationOption
			FileLifetimeOptions []fileLifetimeOption
		}{
			commonProps: makeCommonProps("PicoShare - New Guest Link", r.Context()),
			ExpirationOptions: []expirationOption{
				{"1 day", time.Now().AddDate(0, 0, 1), false},
				{"7 days", time.Now().AddDate(0, 0, 7), false},
				{"30 days", time.Now().AddDate(0, 0, 30), false},
				{"1 year", time.Now().AddDate(1, 0, 0), false},
				{"Never", time.Time(picoshare.NeverExpire), true},
			},
			FileLifetimeOptions: []fileLifetimeOption{
				{picoshare.NewFileLifetimeInDays(7), false},
				{picoshare.NewFileLifetimeInDays(30), false},
				{picoshare.NewFileLifetimeInYears(1), false},
				{picoshare.FileLifetimeInfinite, true},
			},
		}); err != nil {
			http.Error(w, err.Error(), http.StatusInternalServerError)
			return
		}
	}
}
```

We don't need a `FriendlyName` field because we can use the `FileLifetime.FriendlyName` function.

Reply:
This was exactly what I was missing to make it all work! Thanks!